File: src/main/java/com/team3/gudit/purchase/exception/PurchaseErrorCodeProvider.java
package com.team3.gudit.purchase.exception;

import com.team3.gudit.global.exception.ErrorCode;
import com.team3.gudit.global.exception.ErrorCodeProvider;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class PurchaseErrorCodeProvider implements ErrorCodeProvider<PurchaseErrorCode> {
    @Override
    public String getDomain() {
        return "PURCHASE";
    }

    @Override
    public List<ErrorCode> getErrorCodes() {
        return List.of(PurchaseErrorCode.values());
    }
}

File: src/main/kotlin/com/team3/gudit/purchase/exception/PurchaseErrorCodeProvider.kt
package com.team3.gudit.purchase.exception

import com.team3.gudit.global.exception.ErrorCode
import com.team3.gudit.global.exception.ErrorCodeProvider
import org.springframework.stereotype.Component

@Component
class PurchaseErrorCodeProvider : ErrorCodeProvider<PurchaseErrorCode> {
    override fun getDomain(): String = "PURCHASE"
    override fun getErrorCodes(): List<ErrorCode> = PurchaseErrorCode.entries.toList()
}
